PangleAdapter/src/main/java/com/chartboost/mediation/pangleadapter/PangleAdapter.kt

```diff
@@ -452,63 +477,67 @@ class PangleAdapter : PartnerAdapter {
+            fun resumeOnce(result: Result<PartnerAd>) {
+                if (continuation.isActive) {
+                    continuation.resume(result)
+                }
+            }
 
-                    override fun onError(code: Int, message: String?) {
+            PAGInterstitialAd.loadAd(
+                request.partnerPlacement,
+                PAGInterstitialRequest(),
+                object : PAGInterstitialAdLoadListener {
+                    override fun onError(code: Int, message: String) {
                         PartnerLogController.log(
                             LOAD_FAILED,
                             "Code: $code. Error: $message"
                         )
-                        continuation.resume(
+                        resumeOnce(
                             Result.failure(
                                 ChartboostMediationAdException(ChartboostMediationError.CM_LOAD_FAILURE_UNKNOWN)
                             )
                         )
                     }
 
-                    override fun onFullScreenVideoAdLoad(videoAd: TTFullScreenVideoAd?) {
-                        fullScreenAd = videoAd
-                    }
-
-                    override fun onFullScreenVideoCached() {
-                        fullScreenAd?.let {
-                            PartnerLogController.log(LOAD_SUCCEEDED)
-                            continuation.resume(
-                                Result.success(
-                                    PartnerAd(
-                                        ad = it,
-                                        details = emptyMap(),
-                                        request = request
-                                    )
-                                )
+                    override fun onAdLoaded(pagInterstitialAd: PAGInterstitialAd?) {
+                        pagInterstitialAd?.let {
+                            val partnerAd = PartnerAd(
+                                ad = it,
+                                details = emptyMap(),
+                                request = request,
                             )
+                            it.setAdInteractionListener(object :
+                                PAGInterstitialAdInteractionListener {
+                                override fun onAdShowed() {
+                                    onShowSuccess()
+                                }
+
+                                override fun onAdClicked() {
+                                    PartnerLogController.log(DID_CLICK)
+                                    partnerAdListener.onPartnerAdClicked(partnerAd)
+                                }
+
+                                override fun onAdDismissed() {
+                                    PartnerLogController.log(DID_DISMISS)
+                                    partnerAdListener.onPartnerAdDismissed(partnerAd, null)
+                                }
+                            })
```

Comment:
Can we set the listener in _our_ show method instead? That way we won't need to use the `onShowSuccess()` lambda, and can just resume the continuation like we normally would.

Reply:
Yeah, we can. 👍🏼 Let me give it a go.